Move of logs out of apiserver and into component-base

Klaven opened the pull request:
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:
This is a continuation of the component-base started in this https://github.com/kubernetes/kubernetes/pull/72569 PR. This PR moves the `k8s.io/apiserver/pkg/util/logs` pkg to `k8s.io/component-base/cli/logs` as detailed in the component-base backlog as outlined by https://github.com/kubernetes/community/pull/3008.


**Which issue(s) this PR fixes**:
<!-- 
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:
this is a PR directly regarding the above working group.
**Does this PR introduce a user-facing change?**:
<!--  
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

Klaven commented:
/assign @luxas

fedebongio commented:
/cc @cheftako

luxas commented:
/assign @luxas @sttts @mtaufen 
/ok-to-test
/milestone v1.14
/priority backlog

luxas reviewed (approved):
/approve
Thanks @Klaven! This looks good to me on a high level :+1:

I think we need to update a few import restriction files etc. so I'll wait with the LGTM until all tests are passing.

/assign @dims 
who have been working on this stuff as well.

/assign @thockin 
for approval. With this effort we hope to fixup the TODO from PR https://github.com/kubernetes/kubernetes/pull/240 soon-ish :wink:

Klaven commented:
I will fix the verify imports. Let me know if you want me to change the directory structure and I will get it updated.

dims commented:
+1 to `component-base/logging` @luxas (Naming is hard!!!)

Klaven commented:
consider it done. as soon as I get the time I will have this updated and pushed.

Klaven commented:
/retest

Klaven commented:
per a conversation we had on slack. we went with `/component-base/logs` due to naming.

Klaven commented:
/retest

luxas commented on cmd/kube-controller-manager/controller-manager.go:
```diff
@@ -26,8 +26,8 @@ import (
 
-	"k8s.io/apiserver/pkg/util/logs"
 	"k8s.io/kubernetes/cmd/kube-controller-manager/app"
+	"k8s.io/kubernetes/pkg/kubectl/util/logs"
```
??

Klaven replied:
yeah... I have no idea. I will fix this.

luxas commented on hack/import-restrictions.yaml:
```diff
@@ -94,6 +94,7 @@
   - k8s.io/kube-openapi
   - k8s.io/utils
   - k8s.io/klog
+  - k8s.io/component-base
```
nit: add this in alphabetical order (I realize klog violate this, but anyway...). For all of these.

luxas reviewed (commented):
Thanks for the update!
With regards to commit structure for refactors, please follow this pattern:
 - code changes (in one or many commits, here one only is necessary)
 - policy file updates (in this case; import restrictions)
 - autogenerated files (bazel and godeps)

That makes it way easier for the reviewer to wrap their head around the PR once they grow even larger  than this.

Klaven commented:
> Thanks for the update!
> With regards to commit structure for refactors, please follow this pattern:
> 
>     * code changes (in one or many commits, here one only is necessary)
> 
>     * policy file updates (in this case; import restrictions)
> 
>     * autogenerated files (bazel and godeps)
> 
> 
> That makes it way easier for the reviewer to wrap their head around the PR once they grow even larger than this.

@luxas do you want me to make a new PR with it done like this? or is this for future commits like this?

luxas commented:
> @luxas do you want me to make a new PR with it done like this? or is this for future commits like this?

Do this in this PR please. Thanks!

Commit:
move logs into component-base

all the code changes to move move /apiserver/pkg/util/logs into /component-base/logs

Commit:
policy file updates

updated import-restrictions to allow the new component-base/logs if they previously used it before the move.

Commit:
autogen files

all the autogen files that got updated due to the changes to move logs.

Klaven commented:
/retest pull-kubernetes-local-e2e-containerized

Klaven commented:
/retest

luxas commented:
/retest

luxas reviewed (approved):
/lgtm

thockin commented:
/lgtm
/approve
